### DISCUSSION TITLE
localsocket as a two-way on windows

### DISCUSSION THREAD

**Discussion by KawaiiSelbst** [kotauskas/interprocess#30] [Q&A] [answered]
Is it possible to use localsocket as a two-way on windows with tokio, or will you have to make 2 sockets for this?

**Comment by kotauskas** [comment, marked as answer]
`LocalSocket` is actually duplex (fancy term for "two-way"/"bidirectional") on all platforms, as is the Tokio version of it, because Unix systems don't support having them be unidirectional. You can split the Tokio ones to perform reads and writes concurrently; update 2.0.0 will add splitting to non-async local sockets as well.

**Reply by KawaiiSelbst** [reply]
Is it possible to receive and send more than one message at a time?

**Reply by kotauskas** [reply]
What do you mean by "at a time" and by "message"? Local sockets only have byte streams for platform compatibility reasons, and Interprocess does not yet provide a message interface wrapper as of 1.2.1.

If you're wondering if you can receive and send simultaneously, you can currently only do that with Tokio by calling `.split()` on the local socket; 2.0.0 will add support for splitting non-Tokio local sockets and implement I/O traits on `&LocalSocket`, allowing you to split them by reference by simply borrowing them.
